src/python/pants/backend/python/tasks2/pex_build_util.py

```diff
@@ -41,10 +41,15 @@ def has_python_requirements(tgt):
 def dump_sources(builder, tgt, log):
   buildroot = get_buildroot()
   log.debug('  Dumping sources: {}'.format(tgt))
-  for relpath in tgt.sources_relative_to_source_root():
+  for relpath in tgt.sources_relative_to_buildroot():
     try:
-      src = os.path.join(buildroot, tgt.target_base, relpath)
-      if isinstance(tgt, Resources):
+      if type(tgt) == Files:
```

Comment:
So if I understand correctly, this means that we want to take paths from `Files` relative to the buildroot but from `Resources` relative to the srcroot?  This could really use a comment explaining the difference between the `isinstance` test in `has_resources` (which will be true for `Resources` targets) and this exact type test (which will not), and why this is correct here.